src/coredata.cs

```diff
@@ -640,25 +707,35 @@ public interface NSManagedObject {
 		void DidTurnIntoFault ();
 
 		[Export ("valueForKey:")]
-		IntPtr ValueForKey (string key);
+		[return: NullAllowed]
+		NSObject ValueForKey (string key);
 
 		[Export ("setValue:forKey:")]
-		void SetValue (IntPtr value, string key);
+		void SetValue ([NullAllowed] NSObject value, string key);
 
 		[Export ("primitiveValueForKey:")]
-		IntPtr PrimitiveValueForKey (string key);
+		[return: NullAllowed]
+		NSObject PrimitiveValueForKey (string key);
 
 		[Export ("setPrimitiveValue:forKey:")]
-		void SetPrimitiveValue (IntPtr value, string key);
+		void SetPrimitiveValue ([NullAllowed] NSObject value, string key);
 
 		[Export ("committedValuesForKeys:")]
-		NSDictionary CommittedValuesForKeys (string[] keys);
+#if XAMCORE_4_0
+		NSDictionary<NSString, NSObject> CommittedValuesForKeys ([NullAllowed] string[] keys);
+#else
+		NSDictionary CommittedValuesForKeys ([NullAllowed] string[] keys);
+#endif
 
 		[Export ("changedValues")]
+#if XAMCORE_4_0
+		NSDictionary<NSString, NSObject> ChangedValues { get; }
+#else
 		NSDictionary ChangedValues { get; }
+#endif
 
 		[Export ("validateValue:forKey:error:")]
-		bool ValidateValue (ref NSObject value, string key, out NSError error);
+		bool ValidateValue (out NSObject value, string key, out NSError error);
```

Comment:
breaking change, better keep `ref` (before `XAMCORE_4_0`) even if `out` is the best definition for the API